File: 6kyu/the_great_digital_conflict.js
function battleCodes(aL, aN) {
  if (!aL || !aN)
    return 'Peace';
  const a = ' abcdefghijklmnopqrstuvwxyz';
  let l_atk, n_atk;
  const l = [], n = [];
  for (let i = 0; i < aL.length; i++)
    l[i] = a.search(aL[i]);
  for (let i = 0; i < aN.length; i++)
    n[i] = Number(aN[i]);
  while (l.length && n.length) {
    l_atk = l[l.length - 1];
    n_atk = n[0];
    n[0] -= l_atk;
    l[l.length - 1] -= n_atk;
    if (l. length > 1) l[l.length - 2] -= n_atk;
    for (let i = l.length - 1; i >= 0; i--)
      if (l[i] < 1) l.splice(i, 1);
    for (let i = n.length - 1; i >= 0; i--)
      if (n[i] < 1) n.splice(i, 1);
  }
  if (!l.length && !n.length)
    return 'Draw';
  else if (l.length > n.length)
    return l.map(x => a[x]).join('');
  else
    return n.join('');
}

File: 6kyu/the_great_digital_conflict.py
def battle_codes(let, num, a='abcdefghijklmnopqrstuvwxyz'):
    if not let or not num:
        return 'Peace'
    al, an = [a.index(c) + 1 for c in let], [int(i) for i in num]
    while al and an:
        la, na = al[-1], an[0]
        try:
            an[0], al[-1], al[-2] = an[0] - la, al[-1] - na, al[-2] - na
        except IndexError:
            an[0], al[-1] = an[0] - la, al[-1] - na
        al, an = [i for i in al if i > 0], [i for i in an if i > 0]
    w = ''.join(str(n) for n in an) if an else ''.join(a[c - 1] for c in al)
    return 'Draw' if al == an else w
